File: python/algorithms/BubbleSort/bubble_sort.py
def asc_bubble_sort(arr):
    i = 0
    swap = True

    while (i < len(arr) - 1) & swap:
        swap = False
        j = len(arr) - 2
        while j >= i:
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swap = True
            j -= 1
        i += 1
    return arr

def des_bubble_sort(arr):
    i = len(arr) - 1
    swap = True

    while i > 0 & swap:
        swap = False
        j = 0
        while j < i:
            if arr[j] < arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swap = True
            j += 1
        i -= 1
    return arr


if __name__ == "__main__":
    main()

File: c/Algorithms/Sort/BubbleSort/bubble_sort.c
/**********************************************
 > File Name: bubble_sort.c
 > Author: Signal Yu
 > Mail: signalfish38861721@gmail.com
 > Time: 11/12/2022
 > Desc:
***********************************************/

#include "bubble_sort.h"
#include <stdio.h>

void bubble_sort(int *arr, int size) {
  int flag = 1;
  for (int i = 0; i < size && flag; i++) {
    flag = 0;
    for (int j = size - 2; j >= i; j--) {
      if (arr[j] > arr[j + 1]) {
        int temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
        flag = 1;
      }
    }
  }
}